File: Reservationpitch.Application/DTOs/StadiumDTOs/Request/StadiumRequestDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Reservationpitch.Application.DTOs.StadiumDTOs.Request
{
    public class StadiumRequestDto
    {
        [Required]
        public string Name { get; set; } = string.Empty;
        public double PriceinHour { get; set; }
        public string Type { get; set; } = string.Empty;
        public string NoOfplayers { get; set; } = null!;
        public Guid stadiumCenterId { get; set; }
    }
}

File: App/stadium_reservation/lib/models/Dtos/stadiumrequestdto.dart
class StadiumRequestDto {
  String name;
  int priceInHour;
  String type;
  String noOfPlayers;
  String stadiumCenterId;

  StadiumRequestDto({
    required this.name,
    required this.priceInHour,
    required this.type,
    required this.noOfPlayers,
    required this.stadiumCenterId,
  });

  // Convert the DTO to a JSON object to send in the POST request
  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'priceinHour': priceInHour,
      'type': type,
      'noOfPlayers': noOfPlayers,
      'stadiumCenterId': stadiumCenterId,
    };
  }
}
